src/manifest.c

```diff
@@ -530,9 +530,12 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		goto out;
 	}
 
-	if (!signature_download_and_verify(url, filename)) {
-		unlink(filename);
-		goto out;
+	if (signing_enabled && !signature_download_and_verify(url, filename)) {
```

Comment:
I would prefer to have swupd-server ship the .signed file in the MoM.tar (which I think it already does), and then we can try reading the local copy first, before attempting an on-the-fly download.